nerfstudio/configs/external_methods.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2022 the Regents of the University of California, Nerfstudio Team and contributors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=invalid-name
+
+"""This file contains the configuration for external methods which are not included in this repository."""
+import subprocess
+import sys
+from dataclasses import dataclass, field
+from typing import Any, Dict, Optional, Tuple, cast
+
+from rich.prompt import Confirm
+
+from nerfstudio.engine.trainer import TrainerConfig
+from nerfstudio.utils.rich_utils import CONSOLE
+
+
+@dataclass
+class ExternalMethod:
+    """External method class. Represents a link to a nerfstudio-compatible method not included in this repository."""
+
+    slug: str
+    model_description: str
+    instructions: str
+    install_commands: Optional[str] = None
```

Comment:
Instead of install command, what if we just have the user specify the git repo

Reply:
Perhaps not all methods in future will be installed from git? Some may be on pip or installed from a git subdirectory? What do you think of the current alternative?

Reply:
What about `pip_package`?